feature-utils/poly-look/src/react-components/filterChips.jsx

```diff
@@ -61,13 +61,9 @@ const FilterChips = ({
         )
   );
 
-  const [initialDefaultChips, setInitialDefaultChips] =
-    useState(defaultActiveChips);
-
-  if (defaultActiveChips !== initialDefaultChips) {
+  useEffect(() => {
     setActiveChips(defaultActiveChips);
-    setInitialDefaultChips(defaultActiveChips);
-  }
+  }, [JSON.stringify(defaultActiveChips)]);
```

Comment:
Is the stringification really needed? If this is the dependence for the effect, same value will always lead to the same string, right?

Reply:
Well, `defaultActiveChips` is already an array of strings. You're substituting an array of strings with an array that contains a single stringified vesion of the same array. If the stringified version of defaultActiveChips changes, it's because it's changed its components (or maybe order, but that's less likely). So it will still produce an effect. Seen from another perspective, I don't see how `defaultActiveChips` would change while `[JSON.stringify(defaultActiveChips)]` would not, or vice versa.

Reply:
Or it changes where it is located in memory, and then JS will think it is a different array.

Reply:
And that's what happens without the stringify. I tried it. 

React thinks it changes on every rerender and we enter an infinite loop.

Reply:
I don't think JS works like that, but fair enough. If you've tested it and it works this way, good work.